polygon-digital-carbon/utils/getRetirementsContractAddress.ts

```diff
@@ -1,10 +1,10 @@
 import { AMOY_KLIMA_CARBON_RETIREMENTS_CONTRACT, KLIMA_CARBON_RETIREMENTS_CONTRACT } from '../../lib/utils/Constants'
-import { Address, BigInt } from '@graphprotocol/graph-ts'
+import { Address, BigInt, Bytes } from '@graphprotocol/graph-ts'
 
 export function getRetirementsContractAddress(network: string): Address {
   return network == 'matic' ? KLIMA_CARBON_RETIREMENTS_CONTRACT : AMOY_KLIMA_CARBON_RETIREMENTS_CONTRACT
 }
 
-export function getC3RetireRequestId(fromToken: Address, index: BigInt): string {
-  return fromToken.toHexString() + '-' + index.toString()
+export function getC3RetireRequestId(fromToken: Address, index: BigInt): Bytes {
+  return fromToken.concatI32(index.toI32())
```

Comment:
hmm so is the idea that we're spending a lot of cycles casting these IDs to strings when they're really just hex?

Reply:
yep. changing string IDs to Bytes is one of their three syncing optimization recommendations. Other two being using block and call handlers as little as possible and having immutable entities (which apparently speeds it up the most but sounds the most error prone).